### PR TITLE
Fixes Issue # 22992: docker commit failing

### LINKED ISSUE #22992: docker commit failing.
I am not able to commit a new container on docker upstream binary.

**Output of `docker version`:**

```
Client:
 Version:      1.12.0-dev
 API version:  1.24
 Go version:   go1.5.4
 Git commit:   a41e823
 Built:        Wed May 25 15:46:25 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0-dev
 API version:  1.24
 Go version:   go1.5.4
 Git commit:   a41e823
 Built:        Wed May 25 15:46:25 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 1
 Running: 0
 Paused: 0
 Stopped: 1
Images: 1
Server Version: 1.12.0-dev
Storage Driver: devicemapper
 Pool Name: docker-8:19-16121939-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 255.5 MB
 Data Space Total: 107.4 GB
 Data Space Available: 107.1 GB
 Metadata Space Used: 753.7 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.109 (2015-09-22)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.4.2-301.fc23.x86_64
Operating System: Fedora 23 (Workstation Edition)
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 11.72 GiB
Name: dhcp-10-19-62-127.boston.devel.redhat.com
ID: M4W2:KN7L:WVHC:KGXZ:ZNAN:CJJJ:5PWS:V7VV:GPHO:65Y3:4UEP:MCUX
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Username: shm32
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Fedora 23

**Steps to reproduce the issue:**
1. docker run -it  fedora /bin/bash
2. Inside the container shell: echo foobar > /home/foobar.txt
3.  docker ps (Get the container ID)
4.  docker commit {container_id} fedora_foobar

**Describe the results you received:**
Error response from daemon: Error: driver devicemapper is returning inconsistent paths for container ace09224bda2df50c9da884ebc5b2d01eed1c39321f73e73f3b02458d8ad2519 ('/var/lib/docker/devicemapper/mnt/824107f221e531a9e402602f5233f7cc9bc64a70dab127643b181c1012a8ab98/rootfs' then '/var/lib/docker/devicemapper/mnt/824107f221e531a9e402602f5233f7cc9bc64a70dab127643b181c1012a8ab98')

**Describe the results you expected:**
container should be committed successfully.

**Additional information you deem important (e.g. issue happens only occasionally):**

### PR DESCRIPTION
**Pull request by shishir-a412ed**
Fixes #22992

### CODE REVIEW

File: daemon/graphdriver/devmapper/driver.go [hunk 1]
```diff
@@ -160,8 +160,9 @@ func (d *Driver) Remove(id string) error {
 // Get mounts a device with given id into the root filesystem
 func (d *Driver) Get(id, mountLabel string) (string, error) {
 	mp := path.Join(d.home, "mnt", id)
+	rootFs := path.Join(mp, "rootfs")
 	if count := d.ctr.Increment(mp); count > 1 {
```
Comment (cpuguy83) [inline daemon/graphdriver/devmapper/driver.go]: Is this the right path being stored in the ref counter?

**Comment by shishir-a412ed** [comment]
What this PR does is:

1) docker create / run / start: this would create a snapshot device and mounts it onto the filesystem.
    So the first time GET operation is called. it will create the `rootfs` directory and return the path to `rootfs`
2) Now when I do docker commit. It will call the GET operation second time. This time the refcount will check that the count > 1 (count=2). so the rootfs already exists, it will just return the path to rootfs.

Earlier it was just returning the mp: `/var/lib/docker/devicemapper/mnt/{ID}` and hence the inconsistent paths error.

Shishir

**Comment by thaJeztah** [comment]
@shishir-a412ed can you add that description as the commit message?

File: daemon/graphdriver/devmapper/driver.go (hunk 1 above)
Comment (shishir-a412ed) [inline daemon/graphdriver/devmapper/driver.go]: @cpuguy83 I think so, `mp: /var/lib/docker/devicemapper/mnt/{ID}`  is still the mountpoint and should be the right path to ref counter.

**Comment by shishir-a412ed** [comment]
@thaJeztah Please check now.

**Comment by rhatdan** [comment]
@rhvgoyal PTAL

**Comment by thaJeztah** [comment]
Thanks @shishir-a412ed! I wonder if we can test this somehow, so that we don't regress in this area, but I'll leave that to @cpuguy83 and @rhvgoyal for now.

**Comment by cpuguy83** [comment]
@thaJeztah I think our CI is not setup right and is currently testing with vfs only.

**Comment by cpuguy83** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**